### DISCUSSION TITLE
How to make a Vertical Cross Section with Pyvista?

### DISCUSSION THREAD

**Discussion by harshamarupudi56** [pyvista/pyvista#2945] [Q&A] [answered]
https://tetgen.pyvista.org/

I'm looking at this example from the documentation. I can see this segment of the code is extracting a part of the mesh horizontally. 
```
# get cell centroids
cells = grid.cells.reshape(-1, 5)[:, 1:]
cell_center = grid.points[cells].mean(1)

# extract cells below the 0 xy plane
mask = cell_center[:, 2] < 0
cell_ind = mask.nonzero()[0]
subgrid = grid.extract_cells(cell_ind)
```

I would like to make a cross section through the object but vertically instead. Split the sphere in half lengthwise.

**Comment by banesullivan** [comment]
the `slice()` filter might be what you're after: 

https://docs.pyvista.org/api/core/_autosummary/pyvista.DataSetFilters.slice.html

```py
import pyvista as pv

mesh = pv.Wavelet()
slc = mesh.slice()

p = pv.Plotter(notebook=0)
p.add_mesh(mesh.outline())
p.add_mesh(slc)
p.show()
```
<img width="624" alt="Screen Shot 2022-07-06 at 1 44 31 PM" src="https://user-images.githubusercontent.com/22067021/177630516-dfe0cdec-953e-4059-9440-c06ae52e9e5c.png">

**Reply by harshamarupudi56** [reply]
Thanks for the response but not quite what I am looking for. mesh.slice() renders the Sphere into a circle rather then creating a cross section of the sphere as seen in the documentation that I linked.

**Reply by akaszynski** [reply]
I think the issue there is `pyvista.Sphere` returns a `pyvista.PolyData`, and slicing a surface will yield a series of line segments.

**Reply by harshamarupudi56** [reply]
Yes that seems to be what's happening here.

**Comment by p-j-smith** [comment, marked as answer]
Hi, I think you need to change the mask you're creating. In your example you're cutting the sphere along the z-axis. To cut along the x-axis:

```py
mask = cell_center[:, 0] > 0
```

And to cut along the y-axis:

```py
mask = cell_center[:, 1] > 0
```
